src/java.base/share/classes/java/lang/foreign/GroupLayout.java

```diff
@@ -82,73 +82,81 @@ public sealed interface GroupLayout extends MemoryLayout permits StructLayout, U
      * of the provided {@link Record} {@code type} by means of matching the names of the
      * record components with the names of the elements in this group layout}
      * <p>
-     * The mapping between the record type and this memory layout is defined as follows;
+     * A mapper {@code M} between a record type {@code R}, a memory layout {@code L} and an {@code offset}
+     * (which is zero for the initial invocation of this method) is defined as follows {@code M(type, layout, offset)};
      * <p>
-     * Let <em>R</em> be the provided record {@code type} with the constituent components
-     * <em>C<sub>0</sub></em>, ..., <em>C<sub>N-1</sub></em>, where <em>N</em> is non-negative.
+     * Let {@code R} comprise its the constituent components
+     * {@code c_1, c_2}, ..., c_N}, where {@code N} is the (non-negative) number of components of {@code R}.
      * <p>
-     * Let <em>L</em> be this group layout with the elements <em>E<sub>0</sub></em>, ..., <em>E<sub>M-1</sub></em>
-     * , where <em>M</em> {@code >=} <em>N</em>.
+     * Let {@code f_a(MemorySegment ms)} be a function that takes
+     * a {@link MemorySegment} {@code ms} and produces a value for {@code c_a}.
      * <p>
-     * Then, for each <em>C<sub>a</sub></em> (<em>a</em> {@code <} <em>N</em>), there must be a corresponding distinct
-     * <em>E<sub>b</sub></em> such that the {@link MemoryLayout#name() name} of <em>C<sub>a</sub></em>
-     * {@link Object#equals(Object) equals} the {@link RecordComponent#getName() name} of the <em>E<sub>b</sub></em> and:
+     * Let {@code L} be a group layout with the elements {@code e_1, e_2, ..., e_M}, where {@code M >= N}.
+     * <p>
+     * Let {@code offsetBetween(GroupLayout layout, MemoryLayout target)} be a function that, via
+     * the method {@link MemoryLayout#byteOffset(PathElement...)}, can compute the offset from the
+     * {@code layout L} to a sub-layout of {@code L}.
+     * <p>
+     * Then, for each {@code c_a, a <= N}, there must be a corresponding distinct
+     * {@code e_b} such that the {@link MemoryLayout#name() name} of {@code c_a}
+     * {@link Object#equals(Object) equals} the {@link RecordComponent#getName() name} of {@code e_b} and:
      * <ul>
      *    <li>
-     *        <h4>If <em>E<sub>b</sub></em> is a {@link ValueLayout };</h4>
-     *        then <em>C<sub>a</sub></em> must be of the exact type of <em>E<sub>b</sub>'s</em> {@link ValueLayout#carrier() carrier()}<br>
-     *        whereby C<sub>a</sub> = f<sub>a</sub>(MemorySegment ms) = ms.get(E<sub>b</sub>, offset) at the appropriate offset.<br>
+     *        <h4>If {@code e_b} is a {@link ValueLayout };</h4>
+     *        then {@code c_a} must be of the exact type of {@code e_b}'s {@link ValueLayout#carrier() carrier()}<br>
+     *        whereby {@code c_a = f_a(MemorySegment ms) = ms.get(e_b, offset + offsetBetween(layout, e_b))}.<br>
```

Comment:
It is not defined what "offset" is. I believe that has to be a parameter to the f_a - e.g. the function takes a segment *and* and offset.